std/net/curl.d

```diff
@@ -669,19 +669,32 @@ if (is(T == char) || is(T == ubyte))
 {
     import std.uri : urlEncode;
 
-    return post(url, urlEncode(postDict), conn);
+    return post!T(url, urlEncode(postDict), conn);
 }
 
 @system unittest
 {
+    import std.algorithm.searching : canFind;
+
+    enum expected = ["name1=value1&name2=value2", "name2=value2&name1=value1"];
+
     foreach (host; [testServer.addr, "http://" ~ testServer.addr])
     {
+        auto postDict = ["name1" : "value1", "name2" : "value2"];
+
+        testServer.handle((s) {
+            auto req = s.recvReq!char;
+            s.send(httpOK(req.bdy));
+        });
+        char[] charRes = post!char(host ~ "/path", postDict);
+        assert(canFind(expected, charRes));
+
         testServer.handle((s) {
             auto req = s.recvReq!char;
             s.send(httpOK(req.bdy));
         });
-        auto res = post(host ~ "/path", ["name1" : "value1", "name2" : "value2"]);
-        assert(res == "name1=value1&name2=value2" || res == "name2=value2&name1=value1");
+        ubyte[] ubyteRes = post!ubyte(host ~ "/path", postDict);
```

Comment:
you could do a static `foreach (T; AliasSeq!(char, ubyte)){...}` to test the two possible accepted `post` template types.